### PR TITLE
dask-gateway & dask-distributed: set SCRATCH_BUCKET env on workers, partially fix dashboard link

### LINKED ISSUE #442: Set SCRATCH_BUCKET env on dask worker pods.
# Summary

When dask-gateway is used to create a dask worker, we don't pass set the SCRATCH_BUCKET environment variable which is probably very relevant also to have set on worker pods.

# User Stories

- I want to initialize/finalize my dask worker work by downloading/uploading data from/to a scratch bucket.

# Acceptance criteria

- SCRATCH_BUCKET and PANGEO_BUCKET environment variables (they have the same value) are set on worker pods also.

# Tasks to complete

- [x] Figure out a way to set the SCRATCH_BUCKET environment variable.
- [x] Do it for JMTE
- [ ] Do it for 2i2c-org/infrastructure (#855)

### PR DESCRIPTION
**Pull request by consideRatio**
This PR automatically sets the `SCRATCH_BUCKET` and `PANGEO_SCRATCH` environment variables on dask-gateway worker pods based on the user pod that spawned them. This can help the users' scripts running on dask-gateway worker pods also make use of these convenience environment variables set on the user pods.

This PR also adds a environment variable that configures dask distributed to provide a better suggestion on the dashboard link it sometimes provides. This link otherwise fail to account for the JupyterHub prefix in the path (`/user/<username>`, or `/user/<username>/<servername>` for named servers). I've made a note that this adjustment only works for non-named servers, and I don't have a suggestion to make it work for the named servers as well.

This PR has been demonstrated to function well in the JMTE hub, and with JupyterHub 1.1.3 on all hubs, this can now work for all the infrastructure hubs.

Closes #442 - Note that this PR doesn't include environment variables seen in that issue about configuring the dask-labextension to create dask-gateway cluster's via the buttons because that would be a breaking change and it didn't work well in general to do so in JMTE.

### CODE REVIEW

**Review by damianavila — APPROVED** [review]
LGTM.
In addition, I truly appreciate all the inline comments, they are super useful to understand the whole context!

**Comment by consideRatio** [comment]
Thanks @damianavila for your review and encouraging comment! I struggled a lot to understand this.

I'm afraid of doing merges in this repo as I'm kind of expecting some intermittent failure to show up, making me feel responsible to debug to ensure this didn't make some deployment enter a problematic state. If someone would like to merge this, please do - if I'm to do it, I'd like to leave it unmerged for some months hoping we've had time to make the CI system more robust to not fail when it shouldn't.

Instead of merging this and taking such risk of intermittent CI failure, I'll go work on solving #279 that helps us in the right direction towards making our CI system more robust.

**Review by yuvipanda — APPROVED** [review]
<3

**Comment by consideRatio** [comment]
I'll go for a merge as I consider #279 to be blocked by a refactoring to make helm chart values be standalone files currently, and I dislike having stale PRs open.

**Pull request merged**
